app/src/main/java/se/sigmaconnectivity/blescanner/service/BleScanService.kt

```diff
@@ -119,7 +118,12 @@ class BleScanService() : Service() {
             .setCallbackType(ScanSettings.CALLBACK_TYPE_FIRST_MATCH or ScanSettings.CALLBACK_TYPE_MATCH_LOST)
             .build()
         val scanFilter = ScanFilter.Builder()
-           .setServiceUuid( ParcelUuid(UUID.fromString(Consts.SERVICE_UUID)))
+        //TODO: There is a problem with filter on some devices, resolve it and restore filter
+           //            .setServiceData(
+            //                ParcelUuid(UUID.fromString(Consts.SERVICE_UUID)),
+            //                ByteBuffer.allocate(8).array(),
+            //                ByteBuffer.allocate(8).array()
+            //            )
```

Comment:
The main problem is in above with 
`.setCallbackType(ScanSettings.CALLBACK_TYPE_FIRST_MATCH or ScanSettings.CALLBACK_TYPE_MATCH_LOST)`
just change it to `.setCallbackType(ScanSettings.CALLBACK_TYPE_ALL_MATCHES)` an this will do the thing.